gammapy/datasets/tests/test_map.py

```diff
@@ -755,6 +755,7 @@ def test_prior_stat_sum(sky_model, geom, geom_etrue):
 
     uniformprior = UniformPrior(min=0, max=np.inf, weight=1)
     datasets.models.parameters["amplitude"].prior = uniformprior
+    assert_allclose(datasets._stat_sum_likelihood(), 12825.9370, rtol=1e-3)
     assert_allclose(datasets.stat_sum(), 12825.9370, rtol=1e-3)
 
```

Comment:
as `(datasets._stat_sum_likelihood()` is equal to `datasets.stat_sum()`, why is this test function really testing the prior stat_sum? 
Should we not use a prior that lead to a non-nul prior_stat_sum?

Reply:
Here it is the same because it tests a uniform prior with weight=1,
at the end of the test there is also a check with weight=100 that actually make a difference.